### PR TITLE
[AMDGPU] Enable AMDGPUAttributorPass in full LTO

### PR DESCRIPTION
**Pull request by shiltian**
This is basically same as https://github.com/llvm/llvm-project/pull/102086 but reverts some test case changes that are no longer needed.

### CODE REVIEW

**Comment by shiltian** [comment]
* **#102673** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/102673?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @shiltian and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by shiltian** [comment]
This PR passed the internal PSDB.

**Comment by shiltian** [comment]
ping

File: llvm/test/CodeGen/AMDGPU/print-pipeline-passes.ll [hunk 1]
```diff
@@ -0,0 +1,10 @@
+; RUN: opt -mtriple=amdgcn--amdhsa -S -passes="lto<O1>" -print-pipeline-passes %s -o - | FileCheck %s
+; RUN: opt -mtriple=amdgcn--amdhsa -S -passes="lto<O2>" -print-pipeline-passes %s -o - | FileCheck %s
+; RUN: opt -mtriple=amdgcn--amdhsa -S -passes="lto<O3>" -print-pipeline-passes %s -o - | FileCheck %s
+
```
Comment (arsenm) [inline llvm/test/CodeGen/AMDGPU/print-pipeline-passes.ll:4]: Maybe check it's not there for -O0

**Review by arsenm — APPROVED** [review]

**Comment by ye-luo** [comment]
It seems broke https://github.com/llvm/llvm-project/issues/104397

**Pull request merged**